analyzers/tests/SonarAnalyzer.UnitTest/Wrappers/TypeInfoExtensionsTest.cs

```diff
@@ -0,0 +1,158 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2022 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using StyleCop.Analyzers.Lightup;
+using NullabilityInfo = StyleCop.Analyzers.Lightup.NullabilityInfo;
+using NullableAnnotation = StyleCop.Analyzers.Lightup.NullableAnnotation;
+using NullableFlowState = StyleCop.Analyzers.Lightup.NullableFlowState;
+
+namespace SonarAnalyzer.UnitTest.Wrappers
+{
+    [TestClass]
+    public class TypeInfoExtensionsTest
```

Comment:
We generally don't test shimlayer. Just make sure it works in the PR that is based on this and delete the file.

Reply:
There 3 other test files next to this one that test other shim layer types. If we keep our own expression tree, we need the test for debugging as Expression trees are hard to construct otherwise.